### ISSUE TITLE
Ambiguous method call with optional parameters and overload

### ISSUE THREAD

**Issue by TheOpenDevProject** [dotnet/roslyn#11648]
**Version Used**: 
Microsoft Visual Studio Enterprise 2015
Version 14.0.25029.00 Update 2 RC
Microsoft .NET Framework
Version 4.6.01055
**Steps to Reproduce**:

1. Create a class like so
```C#
public class MyClass {
 public override IEnumerable < T > RetrieveAll(string PrimaryKey = null) {
  SqlCommand cmd = new SqlCommand() {
   CommandType = CommandType.StoredProcedure,
    CommandText = ""
  };

 }


 public override IEnumerable < T > RetrieveAll() {
  throw new NotImplementedException();
 }
}
```
2. Make call to method RetrieveAll() << Notice how the call is ambiguous because we did not pass any parameters in.
3. No compile errors are thrown. 

**Expected Behavior**:
Compiler should warn or halt compilation due to ambiguous call.
**Actual Behavior**:
Compiles fine and calls the overload without a parameter.

Does C# explicitly define some sort of order of precedence in this situation about which one to call?

**Comment by jaredpar** [comment]
Thanks for taking the time to report this issue. This behavior is "By Design".  There is a tie breaking rule to sort out this very problem.  For the spec under **Better Function** in **Named and Optional Parameters"

> As a tie breaker rule, a function member for which all arguments where explicitly given is better than one for which default values were supplied in lieu of explicit arguments. 

In this case the method with no arguments wins because all of the arguments, in this case none, were supplied explicitly.
